_data-prepper/pipelines/configuration/sinks/s3.md

```diff
@@ -119,72 +150,50 @@ Option | Required | Type | Description
+When you provide your own Avro schema, that schema defines the final structure of your data. Therefore, any extra values inside any incoming events that are not mapped in the Arvo schema will not be included in the final destination. To avoid confusion between a custom Arvo schema and the `include_keys` or `exclude_keys` sink configurations, Data Prepper does not allow the use of the `include_keys` or `exclude_keys` with a custom schema.
 
+In cases where your data is uniform, you may be able to automatically generate a schema. Automatically generated schemas are based on the first event that the codec receives.
+The schema will only contain keys from this event. Therefore, you must have all keys present in all events in order for the automatically generated schema to produce a working schema. Automatically generated schemas make all fields nullable. Use the sink's `include_keys` and `exclude_keys` configurations to control what data is included in the auto-generated schema.
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
 `schema` | Yes | String | The Avro [schema declaration](https://avro.apache.org/docs/current/specification/#schema-declaration). Not required if `auto_schema` is set to true.
 `auto_schema` | No | Boolean | When set to `true`, automatically generates the Avro [schema declaration](https://avro.apache.org/docs/current/specification/#schema-declaration) from the first event.
-
  
 ### ndjson codec
 
-The `ndjson` codec writes each line as a JSON object.
-
-The `ndjson` codec does not take any configurations.
-
+The `ndjson` codec writes each line as a JSON object. The `ndjson` codec does not take any configurations.
 
 ### json codec
 
-The `json` codec writes events in a single large JSON file.
-Each event is written into an object within a JSON array.
-
+The `json` codec writes events in a single large JSON file. Each event is written into an object within a JSON array.
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
 `key_name` | No | String | The name of the key for the JSON array. By default this is `events`.
 
-
 ### parquet codec
 
-The `parquet` codec writes events into a Parquet file.
-When using the Parquet codec, set the `buffer_type` to `in_memory`.
+The `parquet` codec writes events into a Parquet file. When using the codec, set `buffer_type` to `in_memory`.
 
-The Parquet codec writes data using the Avro schema.
-Because Parquet requires an Avro schema, you may either define the schema yourself, or Data Prepper will automatically generate a schema.
-However, we generally recommend that you define your own schema so that it can best meet your needs.
+The Parquet codec writes data using the  schema. Because Parquet requires an Avro schema, you may either define the schema yourself, or Data Prepper will automatically generate a schema. However, we generally recommend that you define your own schema so that it can best meet your needs.
```

Comment:
```suggestion
The `parquet` codec writes data using the schema. Because Parquet requires an Avro schema, you may either define the schema yourself or have Data Prepper automatically generate it. It is recommended that you define your schema to accurately reflect your needs.
```